src/metabase/models/card.clj

```diff
@@ -78,4 +80,44 @@
   revision/IRevisioned
   {:serialize-instance serialize-instance
    :revert-to-revision revision/default-revert-to-revision
-   :describe-diff      revision/default-describe-diff})
+   :diff-map           revision/default-diff-map
+   :diff-str           revision/default-diff-str})
+
+
+;;; ## ---------------------------------------- DEPENDENCIES ----------------------------------------
+
+
+(defn- parse-filter-subclause [subclause]
+  (match subclause
+    ["SEGMENT" (segment-id :guard integer?)] segment-id
+    subclause                                nil))
+
+(defn- parse-filter [clause]
+  (match clause
+    ["AND" & subclauses] (mapv parse-filter subclauses)
+    ["OR" & subclauses]  (mapv parse-filter subclauses)
+    subclause            (parse-filter-subclause subclause)))
+
+(defn- extract-segment-ids [filter-clause]
+  (when filter-clause
+    (->> (parse-filter filter-clause)
+         flatten
+         (filter identity)
+         set)))
+
+(defn- extract-metric-ids [aggregation-clause]
+  (match aggregation-clause
+    ["METRIC" (metric-id :guard integer?)] #{metric-id}
```

Comment:
Every other aggregation clause is lowercased. `rows`, `count`, `avg`, etc.

I'm probably just going to make the QL keywords case-insensitive (and `lisp-case` vs. `snake_case` insensitive too) in the near future, but until then we should try to hold on to what little consistency we have